static/app/components/codecov/virtualRenderers/lineNumber.tsx

```diff
@@ -54,8 +53,28 @@ export function LineNumber({
   );
 }
 
+// This function generates a pseudo-element that is used to highlight the line number.
+// We need to do this because the background color for the highlights has an alpha
+// channel, so when a user clicks on it, it mixes with the ColorBar background. Because
+// of this, we manually set the main elements background color to that of the code block.
+function generatePseudoElement({background}: {background: string}) {
+  return css`
+    &::before {
+      content: '';
+      position: absolute;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: 100%;
+      z-index: -1;
+      background-color: ${background};
+    }
+  `;
+}
+
```

Comment:
Had an interesting problem with the `LineNumber` component where I had to bring in a pseudo element to resolve my issue by overlaying it and applying the highlighting color.

Reply:
So since highlights background color has an alpha, when I click it there's a bit of transparency and I see the bg color of the ColorBar? So you created a pseudo-element that doesn't have an alpha channel, and when you click the line, then there's no transparency?

Reply:
So what i did was set the non-pseudo-element to have the same background colour as the code block, and then the pseudo-element has the background colour on it, which overlays the element that has that set background colour.